modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java

```diff
@@ -185,11 +200,40 @@ private void ping(WebSocketSession ws) {
      * @throws IOException If failed to process.
      */
     private void registerAgent(WebSocketEvent evt, WebSocketSession ws) throws IOException {
-        AgentInfo agentInfo = fromJson(evt.getPayload(), AgentInfo.class);
+        AgentHandshakeRequest req = fromJson(evt.getPayload(), AgentHandshakeRequest.class);
+
+        AgentHandshakeResponse res = new AgentHandshakeResponse();
```

Comment:
I think will be better to create separate constructor or builder method for error and response.

Reply:
Fixed.